packages/policy/src/changer.ts

```diff
@@ -55,108 +51,95 @@ async function cachedGet(url: string) {
- * @param owner Repostiory owner - ex: "googleapis"
- * @param repo Repository name - ex: "repo-automation-bots"
- * @param octokit Pre-authenticated octokit instance
- */
-export async function addCodeOfConduct(repo: GitHubRepo, octokit: Octokit) {
-  // first, make sure there's no open PR for this
-  const title = 'chore: add a Code of Conduct';
-  const prs = await octokit.search.issuesAndPullRequests({
-    q: `repo:${repo.full_name} "${title}" in:title is:open`,
-  });
-  if (prs.data.total_count > 0) {
-    return;
+export class Changer {
+  private octokit: Octokit;
+  private repo: GitHubRepo;
+  private owner: string;
+  private name: string;
+  private prsCache?: PullRequests;
+
+  constructor(octokit: Octokit, repo: GitHubRepo) {
+    this.octokit = octokit;
+    this.repo = repo;
+    const [owner, name] = repo.full_name.split('/');
+    this.owner = owner;
+    this.name = name;
   }
 
-  // fetch the CoC from `googleapis/.github`
-  const content = await getCoC();
-
-  // submit the PR
-  const changes: Changes = new Map([
-    [
-      'CODE_OF_CONDUCT.md',
-      {
-        content,
-        mode: '100644',
-      },
-    ],
-  ]);
-
-  const [owner, name] = repo.full_name.split('/');
-  await createPullRequest(octokit, changes, {
-    title,
-    message: title,
-    description: 'add a code of conduct',
-    upstreamOwner: owner,
-    upstreamRepo: name,
-    fork: false,
-    primary: repo.default_branch,
-    retry: 0,
-    branch: `policy-bot-${uuid()}`,
-  });
-}
-
-/**
- * Submit a pull request to the target repository to add a SECURITY.md.
- * @param repo Repository name - ex: "repo-automation-bots"
- * @param octokit Pre-authenticated octokit instance
- */
-export async function addSecurityPolicy(repo: GitHubRepo, octokit: Octokit) {
-  // first, make sure there's no open PR for this
-  const title = 'chore: add SECURITY.md';
-  const prs = await octokit.search.issuesAndPullRequests({
-    q: `repo:${repo.full_name} "${title}" in:title is:open`,
-  });
-  if (prs.data.total_count > 0) {
-    return;
+  /**
+   * Check to see if the current repository has an open pull request with the given title
+   * @param title The title to search for
+   * @returns
+   */
+  private async hasMatchingPR(title: string) {
+    if (!this.prsCache) {
+      this.prsCache = await this.octokit.paginate(this.octokit.pulls.list, {
+        owner: this.owner,
+        repo: this.name,
+        state: 'open',
+        per_page: 100,
+      });
+    }
+    const matches = this.prsCache.filter(x => x.title === title);
+    return matches.length > 0;
   }
 
-  // fetch the SECURITY.md from `googleapis/.github`
-  const content = await getSecurity();
-
-  // submit the PR
-  const changes: Changes = new Map([
-    [
-      'SECURITY.md',
-      {
-        content,
-        mode: '100644',
-      },
-    ],
-  ]);
+  /**
+   * Submit a pull request to the target repository to add a SECURITY.md.
```

Comment:
This could be a SECURITY.md or a CODE_OF_CONDUCT.md? perhaps just rephrase:

`Submit a pull request to the target repository to add files required to meet policy, e.g., SECURITY.md`